tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/restrict_policies_test.py

```diff
@@ -0,0 +1,583 @@
+
+      sess.run(
+          var.restrict_policy.apply_restriction(len(updated_features),
+                                                trigger=len(updated_features)))
+      for v in all_vars:
+        self.assertAllEqual(sess.run(v.size()), len(updated_features))
+      keys, _ = sess.run(var.export())
+      self.assertAllEqual(keys, updated_features)
+
+
+class FrequencyRestrictPolicyV1Test(test.TestCase, RestrictPolicyV1TestBase):
+
+  def commonly_apply_update_verify(self):
+    first_inputs = np.array(range(3), dtype=np.int64)
+    second_inputs = np.array(range(1, 4), dtype=np.int64)
+    overdue_features = np.array([0, 3], dtype=np.int64)
+    updated_features = np.array(range(1, 3), dtype=np.int64)
+    with session.Session(config=default_config) as sess:
+      ids = array_ops.placeholder(dtypes.int64)
+      var = de.get_variable('sp_var',
+                            key_dtype=ids.dtype,
+                            value_dtype=dtypes.float32,
+                            initializer=-0.1,
+                            dim=2)
+      embed_w, trainable = de.embedding_lookup(var, ids, return_trainable=True)
+      policy = de.FrequencyRestrictPolicy(var)
+      update_op = policy.apply_update(ids)
+
+      self.assertAllEqual(sess.run(policy.status.size()), 0)
+      sess.run(update_op, feed_dict={ids: first_inputs})
+      self.assertAllEqual(sess.run(policy.status.size()), 3)
+      time.sleep(1)
+      sess.run(update_op, feed_dict={ids: second_inputs})
+      self.assertAllEqual(sess.run(policy.status.size()), 4)
+
+      keys, freq = sess.run(policy.status.export())
+      kvs = sorted(dict(zip(keys, freq)).items())
+      freq = np.array([x[1] for x in kvs])
+      for x in freq[overdue_features]:
+        for y in freq[updated_features]:
+          self.assertLess(x, y)
+
+  def commonly_apply_restriction_verify(self, optimizer):
+    first_inputs = np.array(range(6), dtype=np.int64)
+    second_inputs = np.array(range(4, 9), dtype=np.int64)
+    overdue_features = np.array([0, 1, 2, 3, 6, 7, 8], dtype=np.int64)
+    updated_features = np.array(range(4, 6), dtype=np.int64)
+    all_input_features = np.array(range(9), dtype=np.int64)
+    embedding_dim = 2
+    oversize_trigger = 100
+    optimizer = de.DynamicEmbeddingOptimizer(optimizer)
+
+    with session.Session(config=default_config) as sess:
+      ids = array_ops.placeholder(dtypes.int64)
+      var = de.get_variable('sp_var',
+                            key_dtype=ids.dtype,
+                            value_dtype=dtypes.float32,
+                            initializer=-0.1,
+                            dim=embedding_dim,
+                            restrict_policy=de.FrequencyRestrictPolicy)
+      embed_w, trainable = de.embedding_lookup(var, ids, return_trainable=True)
+      loss = _simple_loss(embed_w)
+      train_op = optimizer.minimize(loss, var_list=[trainable])
+
+      slot_params = [
+          optimizer.get_slot(trainable, name).params
```

Comment:
I cann't see any checking for slots variables' size after restricting.

Reply:
[This](https://github.com/tensorflow/recommenders-addons/pull/27/files#diff-4077cebe6be4ca369d86f091a6e337e530e69cfc2b58dacdc27cb18b813c2cdaR1150-R1163) and [This](https://github.com/tensorflow/recommenders-addons/pull/27/files#diff-4077cebe6be4ca369d86f091a6e337e530e69cfc2b58dacdc27cb18b813c2cdaR1217-R1230) may meet the situation.